azure/scope/cluster.go

```diff
@@ -379,12 +379,15 @@ func (s *ClusterScope) SubnetSpecs() []azure.ResourceSpecGetter {
 			Role:              subnet.Role,
 			NatGatewayName:    subnet.NatGateway.Name,
 		}
+		if s.IsVnetManaged() {
+			subnetSpec.ServiceEndpoints = buildServiceEndpointList(subnet.ServiceEndpoints)
+		}
 		subnetSpecs = append(subnetSpecs, subnetSpec)
 	}
 
 	if s.IsAzureBastionEnabled() {
 		azureBastionSubnet := s.AzureCluster.Spec.BastionSpec.AzureBastion.Subnet
-		subnetSpecs = append(subnetSpecs, &subnets.SubnetSpec{
+		subnetSpec := &subnets.SubnetSpec{
```

Comment:
slight preference for renaming the temporary `subnetSpec` var here to `azureBastionSubnetSpec` to aid the developer's eye in the future when maintaining this as distinct from the above, similar stuff in the `s.AzureCluster.Spec.NetworkSpec.Subnets` enumeration (L369)

Reply:
updated